### PR TITLE
Form Validation

### PR DESCRIPTION
**Pull request by andrupu**
For PXBLUE-1318

Changes proposed in this Pull Request:

- Create the form validation example based on the existing angular design patterns example

### CODE REVIEW

**Comment by emclaug2** [comment]
Can you make sure these unit tests run locally?
CI is failing on test step: https://app.circleci.com/pipelines/github/pxblue/ionic-design-patterns/128/workflows/aae8c8b1-d931-4d1f-ae5f-8857454d445e/jobs/131

File: angular.json [hunk 1]
```diff
@@ -1,187 +1,188 @@
...
-                        "cordovaBuildTarget": "app:ionic-cordova-build",
-                        "devServerTarget": "app:serve"
-                    },
-                    "configurations": {
-                        "production": {
-                            "cordovaBuildTarget": "app:ionic-cordova-build:production",
-                            "devServerTarget": "app:serve:production"
-                        }
-                    }
+  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
```
Comment (joebochill) [inline angular.json:2]: There shouldn't be any changes necessary to angular.json for this. @emclaug2 this file should be covered by our prettier command shouldn't it?

File: README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ This repository holds the source code examples for [PX Blue design patterns](htt
...
 | [Data List](https://pxblue.github.io/patterns/lists)                           | A basic list dynamically built from a JSON data object.                                                                                                                                      | [StackBlitz](https://stackblitz.com/github/pxblue/ionic-design-patterns?file=src%2Fapp%2Fpages%2Flist%2Fdata-list%2Fdata-list.component.html)                               |
+| [Form Validation](https://pxblue.github.io/patterns/lists)                     | Various types of form validation, such as verifying phone number, email address, character length, and password. usage                                                                                                                                    | [StackBlitz]()                               |
```
Comment (joebochill) [inline README.md:16]: Link to the pxblue pattern is pointing to lists instead of form validation.
Missing stackblitz link

**Review by joebochill — CHANGES_REQUESTED** [review]
#### Failing Tests in CI
The tests appear to be failing because the new version of karma returns a fail code when we use invalid command line flags:
These are the command line options that are available:
image

@jeffvg you might need to step in here and help figure out how to do what we need to do with `--code-coverage` and `--no-watch`.

#### Visual Anomolies
When running on a device, the Form Validation page does not look correct. 
- When first navigating to the page, there is a full screen scrim blocking the page. 
- Toolbar has some weird sizing issue, 
- form fields are virtually invisible, 
- etc.:
image

**Comment by jeffvg** [comment]
looks like this branch needs dev merge in. It's missing the reporters in karma config.

reporters: ['progress', 'kjhtml', 'coverage'],

**Comment by joebochill** [comment]
Looks like you got the tests running now, but 8 of them are failing (for the same reason it looks like). Verified that they are failing locally as well.

**Comment by joebochill** [comment]
Looking really good. Couple small things:
- Character counter is not working on the character limit example
- I never get a blue checkmark on the 8 character check in the new password field

Otherwise, 👍

**Comment by emclaug2** [comment]
@andrupu Delete your local yarn.lock file and run `yarn`.  CircleCI has an issue with your current lock file.

**Comment by joebochill** [comment]
One last thing...on the new password field it turns blue when not all of the criteria are met:
image
But it's red if there are fewer than 8 characters entered:
image

According to our design pattern (https://pxblue.github.io/patterns/forms#verifying-on-change), this type of input uses the checks to indicate the verification status, so I think we can remove the error state from the input field entirely.

**Review by joebochill — APPROVED** [review]

**Comment by joebochill** [comment]
Congrats Andrew! Enjoy the green merge button. 🥳

**Comment by jeffvg** [comment]
I'm having issues with android and ios right now....going to clone fresh to see but this is what I have now...

ionic cordova build android
> ng run app:ionic-cordova-build --platform=android
You seem to not be depending on "@angular/core" and/or "rxjs". This is an error.
[ERROR] An error occurred while running subprocess ng.

**Comment by jeffvg** [comment]
something must be wrong on my system. The dev branch at least builds for me unlike the feature branch where it fails on build.
But the dev branch error for me is:
```
cordova build android
Unable to load Platform API from /Users/e0458886/pxblue/ionic-design-patterns/platforms/android/cordova/Api.js:
Cannot find module 'properties-parser'
```
- The Api.js file is there but don't see the properties-parser thingy....

```
native-run android --app platforms/android/app/build/outputs/apk/debug/app-debug.apk
[native-run] No hardware devices found, attempting emulator...
[native-run] Selected emulator emulator-5554
[native-run] Error: ENOENT: no such file or directory, open 'platforms/android/app/build/outputs/apk/debug/app-debug.apk'
[ERROR] An error occurred while running subprocess native-run.

        native-run android --app platforms/android/app/build/outputs/apk/debug/app-debug... exited with exit code 1.
```

**Comment by joebochill** [comment]
Your first issue looks like you have a mix of package-lock and yarn.lock

**Pull request merged**